classes/class-wc-connect-service-settings-store.php

```diff
@@ -103,13 +103,24 @@ public function get_origin_address() {
 			$wc_address_fields = array();
 			$wc_address_fields[ 'company' ] = get_bloginfo( 'name' );
 			$wc_address_fields[ 'name' ] = wp_get_current_user()->display_name;
-			$base_location = wc_get_base_location();
-			$wc_address_fields[ 'country' ] = $base_location[ 'country' ];
-			$wc_address_fields[ 'state' ] = $base_location[ 'state' ];
-			$wc_address_fields[ 'address' ] = '';
-			$wc_address_fields[ 'address_2' ] = '';
-			$wc_address_fields[ 'city' ] = '';
-			$wc_address_fields[ 'postcode' ] = '';
+
+			$wc_countries = WC()->countries;
+			if ( method_exists( $wc_countries, 'get_base_address' ) ) {
```

Comment:
🤔 Do you think this would work better or worse in compatibility class, like `classes/class-wc-connect-compatibility.php` and `classes/class-wc-connect-compatibility-wc30.php`?

Reply:
Yeh, I do agree with the argument that it's not really a compatibility issue. I also think it's a bit convoluted to put in a separate class. Happy to leave it as is :)